### ISSUE TITLE
onChange function is not triggered on React 15.x

### ISSUE THREAD

**Issue by kleberpinel** [wojtekmaj/react-calendar#37]
Hello @wojtekmaj 

I am using your library in a pretty vanila way, but the call back `onChange` is never called when I select something a date in my calendar. Take a look:

![not_working](https://user-images.githubusercontent.com/989051/36288143-21c9fb7c-126e-11e8-8451-9107f66c953d.gif)

Any idea? My version in my package.json is `"react-calendar": "^2.13.0"`

**Comment by sgnl** [comment]
Curious as to what the errors above the "rendering" log?

**Comment by sgnl** [comment]
Try these and see if one of them helps you:

change your `changeSelectedDate` to this syntax:

```javascript
 	changeSelectedDate(date) {
		// your console log here
	}
```

or...

`onChange={() => this.changeSelectedDate}`

or ...

Keeping the code the same but in your `constructor` bind your method to the instance, aka: `this.changeSelectedDate = this.changeSelectedDate.bind(this)`

**Comment by kleberpinel** [comment]
Hey @sgnl, I just tried all of them, none looks to be working.

the `rendering` log is just to show that is passing through that code and is rendering the component.

Any other suggestion ?

**Comment by sgnl** [comment]
Could you post up the example on https://codesandbox.io/ ?

**Comment by sgnl** [comment]
This seems to work: https://codesandbox.io/s/l96wjl78mm

and so does this: https://codesandbox.io/s/l194xo9mq

Do you have any other events on the page? Maybe any css with `pointer-events: none`?

**Comment by jdebarochez** [comment]
I had the same issue. Updating my React dependencies solved the issue. However, I do not explain why it didn't work in the first place.

![image](https://user-images.githubusercontent.com/3436890/36344753-4ad2410c-1438-11e8-9fc6-f96caa8a168a.png)

Thanks @sgnl for the links. It helped found the differences.

**Comment by wojtekmaj** [comment]
Hmmm. There were no breaking changes between React 15.x and 16.x in our case as React did not throw any warnings.

Is the sample I wrote working for you? If not, are there any errors? What do they say? If the sample works, check the differences please! Maybe there's something hard to see for an eye (Babel doing some magic?). React version really shouldn't mattter as long as it's >=15.5.

**Comment by kleberpinel** [comment]
The react version in my project is `15.4.2`.. you just said that it should work do >= 15.5.0 .... Have you tested before for this version (`15.4.2`) ? I forked you project, ran `yarn` and `yarn start` from the `sample` project and I'm getting `react-calendar/sample/Sample.jsx:3:21: Cannot resolve dependency 'react-calendar'`

**Comment by wojtekmaj** [comment]
I did not test it in earlier versions and I don't think they will ever be supported. 15.5 is, at the moment, the oldest version that has a lot in common with the newest versions, mostly because of PropTypes.

**Comment by yummyelin** [comment]
Same issue here. 
Change to using 'onClick', 'onClickDay', still not working.

"react": "^15.5.0",
"react-calendar": "^2.13.1",

**Comment by wojtekmaj** [comment]
All,
I have tested sample and test suites of this repo with multiple versions of React. They definitely do work. If you could share minimal reproduction repo it would be much appreciated.

**Comment by devsli** [comment]
Here the faulty example: https://codesandbox.io/s/2kz06l3p0

The only difference here is React and ReactDOM are of 15.6.1 version

**Comment by wojtekmaj** [comment]
Hmm. This is very odd. I'm calling setState properly and it updates the value internally, but setState callback does not fire.  I filed an issue in React and I hope they will shed some light on it...

facebook/react#12354

**Comment by wojtekmaj** [comment]
@kleberpinel, @devsli,
would you please share your output from `npm list --depth=1`? I have a feeling what might be the problem... Actually you may have React installed twice for some reason, despite React-Calendar's [package.json declaring compatibility with React >=15.5](https://github.com/wojtekmaj/react-calendar/blob/master/package.json#L57-L58)... If that's the case, I know what I can do about it.

**Comment by devsli** [comment]
```
{ ... }
+-- react@15.6.2
| +-- create-react-class@15.6.3 deduped
| +-- fbjs@0.8.16 deduped
| +-- loose-envify@1.3.1 deduped
| +-- object-assign@4.1.1 deduped
| `-- prop-types@15.6.1 deduped
+-- react-addons-test-utils@15.6.0
+-- react-avatar-cropper@0.4.0 invalid
| +-- react@15.6.2 deduped
| +-- react-dom@15.6.2 deduped
| +-- semantic-ui-react@0.68.5 extraneous
| `-- warning@2.1.0 extraneous
+-- react-calendar@2.13.2
| +-- lodash.once@4.1.1
| +-- merge-class-names@1.1.1
| +-- prop-types@15.6.1 deduped
| +-- react@16.2.0
| `-- react-dom@16.2.0
+-- react-calendar-timeline@0.15.9
| +-- element-resize-detector@1.1.14
| +-- interact.js@1.2.8 deduped
| `-- lodash.isequal@4.5.0
+-- react-dom@15.6.2
| +-- fbjs@0.8.16 deduped
| +-- loose-envify@1.3.1 deduped
| +-- object-assign@4.1.1 deduped
| `-- prop-types@15.6.1 deduped
{ ... }
```
Is it enough?

**Comment by wojtekmaj** [comment]
Yes, thank you! Exactly what I suspected. I wrote to npm support on how to handle this, will keep you updated.

**Comment by devsli** [comment]
I think you can try the [peer dependency](https://nodejs.org/en/blog/npm/peer-dependencies/)

**Comment by wojtekmaj** [comment]
Thank you. I am aware of this, but at the same time I need React for development/unit tests, and I'm quite confused on how to handle this, will try today evening.

**Comment by wojtekmaj** [comment]
@devsli, @kleberpinel, @yummyelin 
I believe the issue has been resolved in version 2.13.4. Thank you SO much for your patience. You helped me a lot. Please confirm if the fix works for you - although it does here :) https://codesandbox.io/s/r0989558l4
